Easy use of darkflow as a dependency & Other fixes

abagshaw opened the pull request:
1. I've done a little code consolidation (a little bit of duplicated code between YOLO v1 and YOLO v2 test files) and reorganized the code a little so as to allow for programmic interaction with darkflow (initializing the nets and then passing loaded images to be processed and having the results easily returned).

2. In `./net/yolo/misc.py`,  `model = meta['name']` was setting `model` to the relative file path and the file name and extention, not just the name and extention resulting in the subsequent comparisons to determine the proper label file failing. I changed this to `model = meta['name'].split('/', 1)[-1]` which extracts the actual file name and extention and the comparison completes proplerly. I don't totally understand how something like this has gone unnoticed - so maybe I'm missing something?

~3. In newer versions of TensorFlow, tf.concat() arguments have been reversed. See http://stackoverflow.com/questions/41813665/tensorflow-slim-typeerror-expected-int32-got-list-containing-tensors-of-type I have added a small check to pass the arguments to tf.concat() in the `./net/ops/simple.py` in the correct order depending on the version of TensorFlow that is installed.~
**EDIT:** Whoops, totally messed that up. I was using an older version of TensorFlow and that was causing the issue. No need for any changes (changes have been reverted).

I have tested this with YOLO v2 with both command line interaction as well as with the new direct programmic interaction and everything works fine. Tell me if any changes need to be made.

Thanks!

thtrieu commented:
Cool!

Last time I checked, `tf.concat` is https://www.tensorflow.org/api_docs/python/tf/concat, and the code in this repo is still correct (`axis` being the second argument). We don't think we'll support older version of tensorflow than 1.0

Can you test both v1 and v2 for all four following cases:
- For testing with image output
- For testing with json output
- For testing with camera demo
- Training.

And confirm the result here?

Appreciate a lot :)

abagshaw commented:
Alright - quite a few changes:

1. Ignore point 3 above (see edit) I reverted the changes.

2. Actually funnily enough JSON output was already broken as of 99f9a95 which fixed the bug in the camera demo but also broke the JSON output by declaring two necessary variables past the point at which they were accessed. I fixed that (now both JSON and the camera works)

3. I tested image output, json output, camera demo and programmic interaction for both YOLO v1 and YOLO v2. Everything is working just great (except the results for YOLO v1 are honestly pretty bad...I don't think that's a bug though) out of curiosity, why are you still trying to support YOLO v1?

Unfortunately I don't have time to test training and to be honest I am pretty new with this stuff (I don't really know what I am doing in terms of training). Firstly, I honestly don't think it is necessary to test training as I am 90% sure none of the code I modified has anything to do with the training modules. If you would like training to be tested though - do you think you might have time to do that? I'd love to try to figure out how to do it myself but unfortunately I am going to be away from my computer for the next week and don't have time to do that.

If this could get merged as soon as possible that would be awesome as the code reorganization and consolidation changes could conflict if this is left too long and other changes are made/pull requests merged.

Thanks so much!

thtrieu commented:
It's not that YOLOv1 is bad. It's just that you (accidentally) deleted the non-maximum surpression for v1. I merged the code and added it.
